data/json.c

```diff
@@ -276,7 +276,7 @@ static bool prv_convertValue(_record_t * recordP,
         {
             size_t stringLen;
             uint8_t *string = (uint8_t *)lwm2m_malloc(recordP->valueLen);
-            if (!string) return false;
+            if (string == NULL) return false;
             stringLen = json_unescapeString(string, recordP->value, recordP->valueLen);
             if (stringLen)
```

Comment:
If a NULL comparison for string is added, shouldn't there also be a comparison to 0 for stringLen?

Reply:
No check is really added, only rewritten `if(!string)` <=> `if (string == NULL)` to accommodate #499 concern about "using string as a boolean". I felt it was right to align to the dominating coding style of comparing with NULL.

I see your point but the coding style in the code is more explicit when checking pointers, I believe. Your input on #513 is appreciated